cmake/windows/helpers.cmake

```diff
@@ -0,0 +1,518 @@
+# OBS CMake Windows helper functions module
+
+# cmake-format: off
+# cmake-lint: disable=C0301
+# cmake-lint: disable=R0915
+# cmake-format: on
+
+include_guard(GLOBAL)
+
+include(helpers_common)
+
+# set_target_properties_obs: Set target properties for use in obs-studio
+function(set_target_properties_obs target)
+  set(options "")
+  set(oneValueArgs "")
+  set(multiValueArgs PROPERTIES)
+  cmake_parse_arguments(PARSE_ARGV 0 _STPO "${options}" "${oneValueArgs}" "${multiValueArgs}")
+
+  message(DEBUG "Setting additional properties for target ${target}...")
+
+  while(_STPO_PROPERTIES)
+    list(POP_FRONT _STPO_PROPERTIES key value)
+    set_property(TARGET ${target} PROPERTY ${key} "${value}")
+  endwhile()
+
+  get_target_property(target_type ${target} TYPE)
+
+  if(target_type STREQUAL EXECUTABLE)
+    if(target STREQUAL obs-browser-helper)
+      set(OBS_EXECUTABLE_DESTINATION "${OBS_PLUGIN_DESTINATION}")
+    elseif(target STREQUAL inject-helper OR target STREQUAL get-graphics-offsets)
+      set(OBS_EXECUTABLE_DESTINATION "${OBS_DATA_DESTINATION}/obs-plugins/win-capture")
```

Comment:
Would it be cleaner if the targets could indicate this themselves rather than us handling here? Feel free to wave me off if not worth digging into now.

Does this need an `else` case?

Reply:
Targets have no concept of the application design - that is entirely the concern of the application and the `obs-studio` target and also why any of these "fixups" have been moved into this function (which is also how this is handled on macOS).

Reply:
Got it.